nsqd/nsqd.go

```diff
@@ -370,6 +379,11 @@ func (n *NSQD) PersistMetadata() error {
 		return err
 	}
 
+	err = os.Remove(oldFileName)
```

Comment:
strictly speaking I think we have 3 options w/r/t version promises. We

a) backport a few small deprecation/forwards compatible things that get released as 0.3.9/0.4 and complete removal in 1.0 (documenting the intermediate version requirements for clean transitions) 
b) 1.0 deprecates and provides forwards compatibility w/ removal in 2.0
c) we just cold turkey switch in 1.0 and give strong documentation for the transition risk